File: lib/src/main/java/com/sd/lib/input/stateview/EditTextFocusImageView.java
package com.sd.lib.input.stateview;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.EditText;
import android.widget.ImageView;

import com.sd.lib.input.FEditTextContainer;

/**
 * 焦点变化，切换选中状态
 */
public class EditTextFocusImageView extends ImageView implements FEditTextContainer.UpdateCallback
{
    public EditTextFocusImageView(Context context, AttributeSet attrs)
    {
        super(context, attrs);
    }

    @Override
    public void setVisibility(int visibility)
    {
        if (getVisibility() != visibility)
            super.setVisibility(visibility);
    }

    @Override
    public void setSelected(boolean selected)
    {
        if (isSelected() != selected)
            super.setSelected(selected);
    }

    @Override
    public void onUpdate(EditText editText)
    {
        setSelected(editText.isFocused());
        setVisibility(editText.getVisibility());
    }
}

File: lib/src/main/java/com/sd/lib/input/stateview/EditTextFocusImageView.kt
package com.sd.lib.input.stateview

import android.content.Context
import android.util.AttributeSet
import android.widget.EditText
import android.widget.ImageView
import com.sd.lib.input.FEditTextContainer

/**
 * 焦点变化，切换选中状态
 */
class EditTextFocusImageView : ImageView, FEditTextContainer.StateView {
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)

    override fun setVisibility(visibility: Int) {
        if (this.visibility != visibility) {
            super.setVisibility(visibility)
        }
    }

    override fun setSelected(selected: Boolean) {
        if (this.isSelected != selected) {
            super.setSelected(selected)
        }
    }

    override fun onUpdate(editText: EditText) {
        this.isSelected = editText.isFocused
        this.visibility = editText.visibility
    }
}